ctms/crud.py

```diff
@@ -270,23 +270,33 @@ def get_contacts_by_any_id(
 
 
 def get_all_acoustic_records_before(
-    db: Session,
-    end_time: datetime,
-    retry_limit: int = 5,
+    db: Session, end_time: datetime, retry_limit: int = 5, batch_limit=None
 ) -> List[PendingAcousticRecord]:
     """
     Get all the pending records before a given date. Allows retry limit to be provided at query time."""
-    pending_records: List[PendingAcousticRecord] = (
+    if batch_limit is None:
+        pending_records: List[PendingAcousticRecord] = (
+            db.query(PendingAcousticRecord)
+            .filter(
+                PendingAcousticRecord.update_timestamp < end_time,
+                PendingAcousticRecord.retry < retry_limit,
+            )
+            .order_by(desc(PendingAcousticRecord.update_timestamp))
+            .all()
+        )
+        return pending_records
+    # else
+    batch_pending_records: List[PendingAcousticRecord] = (
         db.query(PendingAcousticRecord)
         .filter(
             PendingAcousticRecord.update_timestamp < end_time,
             PendingAcousticRecord.retry < retry_limit,
         )
-        .order_by(asc(PendingAcousticRecord.update_timestamp))
+        .order_by(desc(PendingAcousticRecord.update_timestamp))
```

Comment:
Is the switch to ``desc`` (last in first out) a permanent thing, or do we want to switch back to ``asc`` (first in first out) eventually?

Reply:
I think it was a temporary thing, given the lack of observability of the system and to unblock testing. I've reverted to asc.